### PR TITLE
Feat/overlay 2

### PR DESCRIPTION
**Pull request by streamich**
This PR touches

- `<Portal>`
- `<Overlay>`
- `@withTheme`
- `<FocusSensor>`
- `<ActiveSensor>`
- `withErrorBoundary()`
- `<CssVarsProvider>`, `<CssVars>`, `withCssVars`
- `<ShouldUpdate>`, `shouldUpdate()`, `pure()`

### CODE REVIEW

File: docs/en/cssvars.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
+# CSS Variables
+
+Use [CSS Custom Properties](https://developer.mozilla.org/en-US/docs/Web/CSS/--*) (aka CSS variables) theming with automatic fallback
+to [regular theming](./theme.md). Below components allow you to safely use CSS variables. You simply use `<CssVarsProvider>` similar
+to how you use [`<Theme>`](./theme.md#theme), but if CSS variables are supported by user's browser, *all values will be replaces instead
```
Comment (xram) [inline docs/en/cssvars.md]: `replaced` not `replaces`.

File: src/ShouldUpdate/__story__/story.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import {linkTo} from '@storybook/addon-links';
+import {ShouldUpdate, shouldUpdate} from '..';
+import ShowDocs from '../../../.storybook/ShowDocs'
+import {Example1} from './Example1';
+import {Example2} from './Example2';
+import {Example3} from './Example3';
+import {Example4} from './Example4';
+
+storiesOf('Inversion/ShouldUpdate', module)
+  // .add('Documentation', () => h(ShowDocs, {md: require('../../../docs/en/ShouldUpdate.md')}))
```
Comment (xram) [inline src/ShouldUpdate/__story__/story.tsx]: Remove?

File: src/cssvars/__story__/story.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+@withCssVars('theme')
+class Decorator1 extends Component<any, any> {
+  render () {
+    return <div style={this.props.theme}>Hello world!</div>;
+  }
+}
+
+storiesOf('Context/CSS Variables', module)
+  // .add('Documentation', () => h(ShowDocs, {md: require('../../../docs/en/FocusSensor.md')}))
```
Comment (xram) [inline src/cssvars/__story__/story.tsx]: Remove?

File: docs/en/cssvars.md (hunk 1 above)
Comment (streamich) [reply docs/en/cssvars.md]: Done.

File: src/ShouldUpdate/__story__/story.tsx (hunk 2 above)
Comment (streamich) [reply src/ShouldUpdate/__story__/story.tsx]: No, uncommented.

File: src/cssvars/__story__/story.tsx (hunk 3 above)
Comment (streamich) [reply src/cssvars/__story__/story.tsx]: No, uncommented.

**Pull request merged**